### PR TITLE
[dev] AB#35112 - Radio Updates and ID Fix

### PR DESCRIPTION
**Pull request by morethanfire**
- Fixed Radio Pill Group id issue.
- Converted SCSS to JSS.

### CODE REVIEW

File: docs/src/inputs/radio/radio.jsx [hunk 1]
```diff
@@ -1,7 +1,6 @@
 import {
```
Comment (groberts314) [inline docs/src/inputs/radio/radio.jsx:1]: I imagine some of the further Documentation modernization is fodder for another day? :-)

File: src/inputs/radio/radio.jsx [hunk 2]
```diff
@@ -54,19 +69,222 @@ const defaultProps = {
...
+                boxShadow: `0 0 0 1px ${theme.palette.active.primary}`,
+            },
+            '& .label': {
+                color: theme.palette.text.primary,
+                cursor: 'pointer',
+                display: 'block',
+                fontSize: 14,
+                position: 'relative',
+                '&::before, &::after': {
+                    content: '""',
```
Comment (groberts314) [inline src/inputs/radio/radio.jsx:129]: Should this just be an empty string (`''`)?  Or is this correct as-is?  Just checking 'cause it looks a little suspicious.  But if it's correct, all good.

File: src/inputs/radio/radio.jsx [hunk 3]
```diff
@@ -138,12 +342,35 @@ class Radio extends React.Component {
...
-         * NOTE: Need to use a prop function here someday.
-         */
+    // onKeyDown() {
+    //     /**
+    //      * NOTE: Need to use a prop function here someday.
+    //      */
 
-        return null;
+    //     return null;
+    // }
```
Comment (groberts314) [inline src/inputs/radio/radio.jsx:351]: Can we just remove this altogether?

File: src/inputs/radio/radio.jsx [hunk 4]
```diff
@@ -231,7 +515,16 @@ class Radio extends React.Component {
                 />
 
                 {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
-                <label className={labelClasses}>
+                <label
```
Comment (groberts314) [inline src/inputs/radio/radio.jsx:518]: Do we deliberately _not_ want to associate the label to an actual control element in this case?

File: src/inputs/radio/radioItem.jsx [hunk 5]
```diff
@@ -1,56 +1,116 @@
...
-        const containerClasses = ClassNames('radio-item', className, {
+
+        const rootClasses = ClassNames('radio-item', className, {
             'radio-item-is-checked': checked,
         });
 
         return (
-            <div className={containerClasses} onClick={this.onClick.bind(this)}>
+            <div
+                aria-checked="false"
```
Comment (groberts314) [inline src/inputs/radio/radioItem.jsx:85]: Does this need to vary depending on value of `checked` prop?

**Review by groberts314 — COMMENTED** [review]
Mostly looks good.  I have a few clarifying questions ... if you have satisfactory answers then I can approve! :-P

File: src/inputs/radio/radio.jsx (hunk 4 above)
Comment (morethanfire) [reply src/inputs/radio/radio.jsx:518]: Yes, for now.

Comment (groberts314) [reply src/inputs/radio/radio.jsx:510]: Sounds good.  Resolving.

File: src/inputs/radio/radio.jsx (hunk 3 above)
Comment (morethanfire) [reply src/inputs/radio/radio.jsx]: Removed.

File: src/inputs/radio/radioItem.jsx (hunk 5 above)
Comment (morethanfire) [reply src/inputs/radio/radioItem.jsx]: Yes, fixed.

File: src/inputs/radio/radio.jsx (hunk 2 above)
Comment (morethanfire) [reply src/inputs/radio/radio.jsx:129]: Correct as is and I agree it is ugly.

File: src/inputs/radio/radio.jsx (hunk 3 above)
Comment (groberts314) [reply src/inputs/radio/radio.jsx]: Looks good.  Resolving.

File: src/inputs/radio/radioItem.jsx (hunk 5 above)
Comment (groberts314) [reply src/inputs/radio/radioItem.jsx]: Looks good.  Resolving.

File: docs/src/inputs/radio/radio.jsx (hunk 1 above)
Comment (morethanfire) [reply docs/src/inputs/radio/radio.jsx:1]: Yeah. I am hoping for 10.0.0 to take care of that.

**Review by groberts314 — APPROVED** [review]

File: src/inputs/radio/radio.jsx (hunk 2 above)
Comment (groberts314) [reply src/inputs/radio/radio.jsx:129]: Copy that.  C'est la vie.  Resolving.

File: docs/src/inputs/radio/radio.jsx (hunk 1 above)
Comment (groberts314) [reply docs/src/inputs/radio/radio.jsx:1]: Resolving.

**Comment by morethanfire** [comment]
May need to thing about the regression risks here. In your opinion @groberts314, would this fix that is slated right now for 9.17.0 be a high or medium regression risk or maybe even low?

**Comment by groberts314** [comment]
@morethanfire Low to medium risk I'd say.  I see potential unforeseen styling regressions as possible, especially in places where we were, for whatever reason, trying to override what was coming in from React CM UI--that's where I think the biggest risk would be present.  But anecdotally, I don't think we don't use radio buttons all _that_ much (we use those "pill" selectors in a few places I guess -- we use traditional radio buttons hardly at all).

But if there's no hurry to get React CM UI 9.17.0 integrated into HC for 1.29.0, and we can hold off until 1.30.0 (we're slated to do release branching like EOD Friday or first thing Monday--we should be "feature complete" by EOD Friday) then I think it's okay.  Even if we hurried it up for 1.29.0, hopefully our regression testing would catch it fast, and we could fix any issues.

**Comment by morethanfire** [comment]
Thank you for your thought on that. I say we release this in 9.17.0 and wait until 1.29.0 is out the door to be implemented into dev over at church-management.

**Comment by groberts314** [comment]
I concur

**Pull request merged**